src/WebJobs.Script/Description/DotNet/PackageManager.cs

```diff
@@ -98,6 +104,135 @@ public static string ResolveNuGetPath()
+            var dependencyGroups = GetDependencyGroups(lockFilePath);
+
+            foreach (var dependencyGroup in dependencyGroups)
+            {
+                IOrderedEnumerable<string> projectDependencies;
+                projectDependencies = projectFrameworks
+                    .FirstOrDefault(f => Equals(dependencyGroup.Framework, f.Framework))
+                    ?.Dependencies?.OrderBy(d => d, StringComparer.OrdinalIgnoreCase);
+
+                if (!projectDependencies?.SequenceEqual(dependencyGroup.Dependencies.OrderBy(d => d, StringComparer.OrdinalIgnoreCase)) ?? false)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static IList<FrameworkInfo> GetProjectDependencies(string projectFilePath)
+        {
+            var frameworks = new List<FrameworkInfo>();
+
+            var jobject = JObject.Parse(File.ReadAllText(projectFilePath));
+            var targetFrameworks = jobject.Value<JToken>("frameworks") as JObject;
+            var projectDependenciesToken = jobject.Value<JToken>("dependencies") as JObject;
+
+            if (projectDependenciesToken != null)
+            {
+                IList<string> dependencies = ReadDependencies(projectDependenciesToken);
+                frameworks.Add(new FrameworkInfo(null, dependencies));
+            }
+
+            if (targetFrameworks != null)
+            {
+                foreach (var item in targetFrameworks)
+                {
+                    var dependenciesToken = item.Value.SelectToken("dependencies") as JObject;
+
+                    var framework = NuGetFramework.Parse(item.Key);
+                    IList<string> dependencies = ReadDependencies(dependenciesToken);
+
+                    frameworks.Add(new FrameworkInfo(framework, dependencies));
+                }
+            }
+
+            return frameworks;
+        }
+
+        private static IList<string> ReadDependencies(JObject dependenciesToken)
+        {
+            var dependencies = new List<string>();
+
+            if (dependenciesToken != null)
+            {
+                foreach (var dependency in dependenciesToken)
+                {
+                    string name = dependency.Key;
+                    string version = null;
+
+                    if (dependency.Value.Type == JTokenType.Object)
+                    {
+                        // { "PackageName" : { "version" :"1.0" ... }
+                        version = dependency.Value.Value<string>("version");
+                    }
+                    else if (dependency.Value.Type == JTokenType.String)
+                    {
+                        // { "PackageName" : "1.0" }
+                        version = dependency.Value.Value<string>();
+                    }
+                    else
+                    {
+                        // Better exception?
```

Comment:
Left as a comment here, but can remove depending on feedback. The exception type used here doesn't seem right...